docs/upgrading.md

```diff
@@ -1,5 +1,118 @@
+implementation loads an OPA policy rule that is compiled to .wasm from a Sawtooth
+address. The `CliPolicyLoader` implementation provides support for loading an OPA
+policy .wasm file via CLI arguments for a file path and policy entrypoint.
+
+Chronicle integrates OPA execution at its GraphQL API request point and in its Transaction
+Processor to ensure access control and policy compliance.
+
+Chronicle's `opa-tp` command-line interface (CLI) is used to interact with the Chronicle
+OPA-TP (Transaction Processor) for Public Key Infrastructure (PKI) and OPA rule storage.
+
+Chronicle now has `opactl`, which allows users to submit transactions to the Sawtooth
+network that specify operations on OPA (Open Policy Agent) policies stored on the
+ledger. The tool provides various commands for querying, listing, and modifying OPA
+policy states on the Sawtooth network.
+
+## No Changes, No Dispatch
+
+We added API methods and responses to check whether Chronicle operations resulting
+from API calls will result in changes in state. If they don't, they won't be submitted
+to the Transaction Processor, instead returning `Submission::AlreadyRecorded` along
+with GraphQL data containing `context` and a `null` `tx_id`.
+
+## In-Domain Documentation Capabilities
+
+Chronicle now provides users with the ability to generate a Chronicle domain with
+documentation by using a domain.yaml file and adding documentation comments to it.
+
+## `async-sawtooth-sdk`
+
+Chronicle now uses `async-sawtooth-sdk` in this update, which contains generic client
+types used by `opa-tp-protocol`. The transition to this new SDK has led to several
+improvements.
+
+The `LedgerReader` and `LedgerWriter` client abstractions are now in common, with
+specialization for transactions and events in the `chronicle-protocol` and `opa-tp-protocol`
+modules. The simulated transaction processors now use the actual Chronicle and OPA
+TransactionHandlers during tests and in `devmode`.
+
+## `prov:wasAttributedTo`
+
+Chronicle now supports the `wasAttributedTo` PROV-O relationship.
+
+## Endpoint offering JSON by IRI
+
+Now, running Chronicle with the `serve-api` command, users can visit
+<http://localhost:9982/data/chronicle:agent:my-agent-id> and see a JSON summary,
+which they can also do for entity and activity.
+
+## To 0.6x
+
+### OPA TP, Protocol, and Skeletal CLI
+
+We introduced an OPA policy and key registration transaction processor in this update.
+The OPA policy and key registration transaction processor share similarities in design
+with the Chronicle TP. Tested at the protobuf level, this feature is designed to
+operate independently of the rest of the Chronicle project.
+
+## To 0.5x
+
+### Move from SQLite to PostgreSQL
+
+Chronicle now utilizes PostgreSQL instead of SQLite as its local database.
+This change brings several benefits, such as increased scalability and better
+performance for large-scale data management.
```

Comment:
As an aside this is noticeable with the ETL on the discover project that's using a fake ledger. Chronicle is blessed performance wise as it is almost certainly dependent on something a lot slower than it so is always going to look good.